### PR TITLE
Fixed example Snippet and clarified it.

### PR DESCRIPTION
**Pull request by sas41**
# Found a possible error in one of the snippets

## Summary

Fixes the 3rd snippet's return type and clarifies the local function's name to emphasize it's status as a local function

## Details

Changed return type of snippet 3 from bool to int, as returned variable was declared as int.

### CODE REVIEW

File: docs/csharp/local-functions-vs-lambdas.md [hunk 1]
```diff
@@ -65,20 +65,20 @@ Third, the compiler can perform static analysis that enables local functions to
...
-    Local();
+    LocalFunction();
     return y;
 
-    void Local() => y = 0;
+    void LocalFunction() => y = 0;
 }
 ```
 
 The compiler can determine that `Local` definitely assigns `y` when called. Because `Local` is called before the `return` statement, `y` is definitiely
```
Comment (pkulikov) [inline docs/csharp/local-functions-vs-lambdas.md]: As `Local` is renamed to `LocalFunction` in the example, please update the paragraph after the example as well.

File: docs/csharp/local-functions-vs-lambdas.md [hunk 2]
```diff
@@ -65,20 +65,20 @@ Third, the compiler can perform static analysis that enables local functions to
...
 assigned at the `return` statement.
 
-The analysis that enables that analysis enables the fourth difference.
+The analysis that enables the example analysis enables the fourth difference.
```
Comment (rpetrusha) [inline docs/csharp/local-functions-vs-lambdas.md]: Even with your clarification of "that", the meaning of this sentence is unclear. I'd suggest removing it, and changing the next sentence to "Fourth, depending..."

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thank you for contributing to the dotnet/docs repo, @sas41, and for clarifying both code and text. I've suggested a change to the paragraph that you modified. Also note the comment from @pkulikov about changing two occurrences of `Local` to `LocalFunction` in the sentence that follows the example.

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
Thank you so much, @sas41, for making the additional change. We'll merge your PR now. Your changes should appear live on docs.microsoft.com in the next day or two.

**Pull request merged**